packages/mobile/src/transactions/saga.ts

```diff
@@ -82,6 +88,62 @@ function* cleanupStandbyTransactions({ transactions }: UpdateTransactionsAction)
   }
 }
 
+function* getInviteTransactionDetails(txHash: string, blockNumber: string) {
+  const kit: ContractKit = yield call(getContractKit)
+  const escrowWrapper: EscrowWrapper = yield call([kit.contracts, kit.contracts.getEscrow])
+  const transferEvents: EventLog[] = yield call(
+    [escrowWrapper, escrowWrapper.getPastEvents],
+    'Transfer',
+    {
+      fromBlock: blockNumber,
+      toBlock: blockNumber,
+    }
+  )
+  const transactionDetails = transferEvents.find(
+    (transferEvent) => transferEvent.transactionHash === txHash
+  )
+
+  if (!transactionDetails) {
+    Logger.error(
+      `${TAG}@getInviteTransactionDetails`,
+      `No escrow past events found with transaction hash ${txHash} and block number ${blockNumber}`
+    )
+    return {}
+  }
+
+  return {
+    recipientIdentifier: transactionDetails.returnValues.identifier,
+    paymentId: transactionDetails.returnValues.paymentId,
+  }
+}
+
+export function* getInviteTransactionsDetails({ transactions }: UpdateTransactionsAction) {
+  const existingInviteTransactions = yield select(inviteTransactionsSelector)
+  const newInviteTransactions = transactions.filter(
+    (transaction) =>
+      transaction.type === TokenTransactionTypeV2.InviteSent &&
+      !existingInviteTransactions[transaction.transactionHash]
+  )
+
+  if (newInviteTransactions.length > 0) {
```

Comment:
nit: it's nice to return early to avoid extra indentation.
```suggestion
  if (newInviteTransactions.length <= 0) { return: }
```

Reply:
that's true, updated